test/common.py

```diff
@@ -2395,8 +2389,27 @@ def log_request(code=0, size=0):
   # do not have the correct MIME type
   SimpleHTTPRequestHandler.extensions_map['.mjs'] = 'text/javascript'
 
-  httpd = ThreadingHTTPServer(('localhost', port), TestServerHandler)
-  httpd.serve_forever() # test runner will kill us
+  return ThreadingHTTPServer(('localhost', port), TestServerHandler)
+
+
+class ServerThread(threading.Thread):
+  """A generic thread class to create and run a server."""
+  def __init__(self, server_creator, *server_args):
+    super().__init__()
+    self.server = None
+    self.server_creator = server_creator
+    self.server_args = server_args
```

Comment:
Can you just do `self.server = server_creator(*server_args)` here, rather then in the run function?

If so I guess you could even accept the full constructed server as the argument instead of the ctor + args?

Reply:
The server needs to be created on a seperate thread which is where `run` is called.